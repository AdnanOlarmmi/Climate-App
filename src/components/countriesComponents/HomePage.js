import React from 'react';
import { Link } from 'react-router-dom';

const HomePage = ({ countriesdata }) => {
  const aqiStatus = (aqi) => {
    if (aqi === 1) {
      return 'Good';
    }
    if (aqi === 2) {
      return 'Moderate';
    }
    if (aqi === 3) {
      return 'Unhealthy';
    }
    if (aqi === 4) {
      return 'Very Unhealthy';
    }

    return 'Hazardous';
  };

  return (
    <main className='homepage flex-row'>
      {countriesdata&&countriesdata.map((countrydata) => (
        <Link className='homepage__item' to={countrydata.ctryinfo.ctryname} key={countrydata.id}>
          <div className='countryname'>{countrydata.ctryinfo.ctryname}</div>
          <div>
            <span>
              Air status:&nbsp;
              {countrydata.data.main.aqi}
            </span>
            {' '}
            <div>{aqiStatus(countrydata.data.main.aqi)}</div>
          </div>
        </Link>
      ))}
    </main>
  );
};

export default HomePage;
