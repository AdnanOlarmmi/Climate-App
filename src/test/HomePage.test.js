import React from 'react';
import render from './App.test';
import HomePage from '../components/countriesComponents/HomePage';

describe('MyRocketss', () => {
  it('render My Rockets Components', () => {
    render(<HomePage />);
    expect(HomePage).toMatchSnapshot();
  });
});
