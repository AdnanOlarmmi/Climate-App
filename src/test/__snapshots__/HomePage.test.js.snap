// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My HomePage render My HomePage Component 1`] = `[Function]`;

exports[`MyRocketss render My Rockets Components 1`] = `[Function]`;
